f5_os_test/heat_client_utils.py

```diff
@@ -0,0 +1,51 @@
+# Copyright 2015-2016 F5 Networks Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+#
+
+import pytest
+
+
+def get_template_file(template_file):
+    file = open(template_file)
+    template_str = file.read()
+    file.close()
+    return template_str
+
+
+def cleanup_stack_if_exists(heat_client, template_name):
+    stacks = heat_client.stacks.list()
+    for stack in stacks:
+        if stack.stack_name == template_name:
+            heat_client.delete_stack(stack.id)
+
+
+@pytest.fixture
+def HeatStack(heatclientmanager, request):
+    '''Fixture for creating/deleting a heat stack.'''
+    def manage_stack(template_file, template_name, parameters={}):
```

Comment:
Is the 'template_name' available from the contents of the template file?
If yes, then extract it from there, else there's a potential conflict between sources of truth.  Also prefer 'ID' to name where possible because of the possibility of naming conflicts in OpenStack (they should be called nicknames).